docs/spec.md

```diff
@@ -4336,18 +4345,23 @@ spaces of `result` and `operand`.
 
 | Label | Name      | Type                       | Constraints |
 |-------|-----------|----------------------------|-------------|
-| (I1)  | `operand` | tensor or quantized tensor | (C1-C2)     |
+| (I1)  | `operand` | tensor or quantized tensor | (C1-C5)     |
 
 #### Outputs
 
 | Name     | Type                       | Constraints |
 |----------|----------------------------|-------------|
-| `result` | tensor or quantized tensor | (C1-C2)     |
+| `result` | tensor or quantized tensor | (C1-C4)     |
 
 #### Constraints
 
-* (C1) `operand` and `result` have the same element type.
-* (C2) `operand` and `result` have the same number of elements.
+* (C1) `operand` and `result` have the same number of elements.
+* If the operation uses non-quantized tensors:
+  * (C2) `operand` and `result` have the same element type.
+* If the operation uses quantized tensors:
+  * (C3) `is_quantized_tensor(operand) and is_quantized_tensor(result)`.
+  * (C4) `quantized_element_type(operand) = quantized_element_type(result)`.
+  * (C5) `is_empty(quantization_dimension(operand))`.
```

Comment:
It would appear that reshaping a per-axis quantized tensor is needed to define semantics of DotGeneralOp per the definition of the op. What do we do about that? (Same question about SliceOp and TransposeOp).

Reply:
That is good point. I would propose to address that as part of https://github.com/openxla/stablehlo/issues/1628. Added to [comment](https://github.com/openxla/stablehlo/issues/1628#issuecomment-1599241549) to make sure we do not miss it.

Reply:
Sounds like a plan.